price-estimator/src/main.rs

```diff
@@ -105,8 +116,17 @@ fn main() {
         options.auction_data_page_size,
         options.orderbook_file,
     );
-    // TODO: Use real price source instead of default.
-    let orderbook = Arc::new(Orderbook::new(Box::new(orderbook), Default::default()));
+
+    let external_price_sources = core::price_estimation::external_price_sources(
+        &http_factory,
+        token_info.clone(),
+        options.price_source_update_interval,
+    )
+    .expect("failed to create external price sources");
+    let infallible_price_source =
+        UpdatingInfalliblePriceSource::new(all_tokens, token_infos, external_price_sources);
```

Comment:
Why does the infallible price source need concrete `TokenInfos` and cannot use a `TokenInfoFetching` trait? In prod we might run for a week straight. Is it right that tokens that are added might unnecessarily fall back to OWL estimates in that case (which might also be hard to debug).

Reply:
Good point. I'll change it.